### PR TITLE
Poor contrast between bg and text when coalition color is light.

### PR DESCRIPTION
**Pull request by raccoman**
I noticed that for coalitions that have a light background color, always using white for the text doesn't create the right contrast.

I decided to implement a simple function in the Stats component that determines which text color should be used based on the brightness of the coalition background color.

### CODE REVIEW

**Comment by JaeSeoKim** [comment]
image

That's a very good idea! 👍 
But, after testing the code, some of the results are not harmonious.

It seems that some code conditions need to be changed.

**Comment by JaeSeoKim** [comment]
I'm working on ways to add shadow effect to characters while solving the "svg safari blurry" issue.

image

File: components/badge/Stats/Stats.tsx [hunk 1]
```diff
@@ -24,17 +24,31 @@ export type StatsProps = {
...
   const height = 190 - (!data.name || !data.email ? 25 : 0);
+  const fgColor = (function() {
+
+    const result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(data.color);
+
+    const r = parseInt(result[1], 16);
+    const g = parseInt(result[2], 16);
+    const b = parseInt(result[3], 16);
+
+    const luminance = 1 - (0.299 * r + 0.587 * g + 0.114 * b) / 255;
```
Comment (GlaceCoding) [inline components/badge/Stats/Stats.tsx]: font-color-contrast algo

https://www.npmjs.com/package/font-color-contrast

**Comment by raccoman** [comment]
Thanks @GlaceCoding for the advice now it should work, maybe the only thing is the white text on the layer bar, but if I change that based on the background too, the black text on the black/transparent background create the same problem of poor contrast.

**Comment by raccoman** [comment]
@JaeSeoKim How it is going?

**Comment by JaeSeoKim** [comment]
@raccoman
I'm sorry for the delay. I couldn't proceed because I was preparing for a job and working on a project.

Currently, SVG image rendering is being reorganized to solve the Poor Contrast problem.

**Comment by raccoman** [comment]
@JaeSeoKim No problem.

To solve the problem of low contrast, are you more inclined to add shadows to the text or to use my solution, in the case of the first I can see if I can give you a hand if you want.

**Comment by JaeSeoKim** [comment]
Currently, we added shadows through the first method.

It seems good to add a function that reverses the shadow color and font color through the method you suggested!
image

**Comment by JaeSeoKim** [comment]
@all-contributors please add @raccoman  for ideas

**Pull request closed without merging**